src/components/avatar/index.tsx

```diff
@@ -222,14 +222,21 @@ class Avatar extends PureComponent<AvatarPropTypes> {
     return _.get(this.props, 'badgeProps.backgroundColor') || statusColor || onlineColor;
   }
 
-  getBadgeSize = () => _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
+  getBadgeSize = () => {
+    const badgeSize = _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
+    
+    if (_.isString(badgeSize)) {
+      return BADGE_SIZES[badgeSize];
```

Comment:
When you split the logic between enum and numeric size you eliminated the fallback of `DEFAULT_BADGE_SIZE` for string-based sizes (currently returns undefined if sending a bad enum). Consider adding `DEFAULT_BADGE_SIZE` fallback if supplied enum size isn't included in `BADGE_SIZES` - to maintain old fallback logic.

Reply:
Right, added.